contracts/PWN.sol

```diff
@@ -41,134 +41,127 @@ contract PWN is Ownable {
-        vault.pullProxy(deed.getOfferLoan(_offer), lender, msg.sender);
-
-        MultiToken.Asset memory collateral;
-        collateral.category = MultiToken.Category.ERC1155;
-        collateral.id = did;
-        collateral.assetAddress = address(deed);
-        vault.pullProxy(collateral, msg.sender, lender);
+    function createDeed(
+        address _collateralAssetAddress,
+        MultiToken.Category _collateralCategory,
+        uint256 _collateralAssetAmount,
+        uint256 _collateralAssetId,
+        address _loanAssetAddress,
+        uint256 _loanAssetAmount,
+        uint256 _loanRepayAmount,
+        uint32 _duration,
+        uint40 _offerExpiration,
+        address _lender,
+        bytes32 _nonce,
+        bytes memory _signature
+    ) external returns (bool) {
+        MultiToken.Asset memory collateral = MultiToken.Asset(
+            _collateralAssetAddress,
+            _collateralCategory,
+            _collateralAssetAmount,
+            _collateralAssetId
+        );
+
+        MultiToken.Asset memory loan = MultiToken.Asset(
+            _loanAssetAddress,
+            MultiToken.Category.ERC20,
+            _loanAssetAmount,
+            0
+        );
+
+        PWNDeed.Offer memory offer = PWNDeed.Offer(
+            collateral,
+            loan,
+            _loanRepayAmount,
+            _duration,
+            _offerExpiration,
+            _lender,
+            _nonce
+        );
+
+        deed.create(offer, _signature, msg.sender);
+
+        vault.push(offer.collateral, msg.sender);
+
+        vault.pullProxy(offer.loan, offer.lender, msg.sender);
 
         return true;
     }
 
     /**
      * repayLoan
-     * @dev the borrower can pay back the funds through this function
-     * @dev the function assumes the asset (and amount to be paid back) to be returned is approved for PWNVault
-     * @dev the function assumes the borrower has the full amount to be paid back in their account
+     * @notice The borrower can pay back the loan through this function
+     * @dev The function assumes the asset (and amount to be paid back) to be returned is approved for PWNVault
+     * @dev The function assumes the borrower has the full amount to be paid back in their account
      * @param _did Deed ID of the deed being paid back
-     * @return true if successful
+     * @return True if successful
      */
     function repayLoan(uint256 _did) external returns (bool) {
         deed.repayLoan(_did);
 
-        bytes32 offer = deed.getAcceptedOffer(_did);
-        MultiToken.Asset memory loan = deed.getOfferLoan(offer);
-        loan.amount = deed.toBePaid(offer);  //override the num of loan given
+        MultiToken.Asset memory loan = deed.getLoan(_did);
+        loan.amount = deed.getLoanRepayAmount(_did);
 
-        vault.pull(deed.getDeedCollateral(_did), deed.getBorrower(_did));
+        vault.pull(deed.getCollateral(_did), deed.getBorrower(_did));
         vault.push(loan, msg.sender);
```

Comment:
change order -> first pay back to prevent unexpected onReceive behavior (the vault should prevent that, but still)

Reply:
Good point. Better safe than sorry.